text/0000-rust-for-linux-project-goal-2024h2.md

```diff
@@ -0,0 +1,238 @@
+
+[Lang]: https://www.rust-lang.org/governance/teams/lang
+[Compiler]: https://www.rust-lang.org/governance/teams/infra
+[Libs-API]: https://www.rust-lang.org/governance/teams/library#team-libs-api
+[nikomatsakis]: https://github.com/nikomatsakis/
+[joshtriplett]: https://github.com/joshtriplett/
+[Team]: https://img.shields.io/badge/Team%20ask-red
+
+# Motivation
+
+The [experimental support for Rust development in the Linux kernel][RFL] is a watershed moment for Rust, demonstrating to the world that Rust is indeed capable of targeting all manner of low-level systems applications. And yet today that support rests on a [number of unstable features][RFL#2], blocking the effort from ever going beyond experimental status. For 2024H2 we will work to close the largest gaps that block support.
+ 
+[RFL]: https://rust-for-linux.com/
+[RFL#2]: https://github.com/Rust-for-Linux/linux/issues/2
+
+## The status quo
+
+The [Rust For Linux (RFL)][RFL] project has been accepted into the Linux kernel in experimental status. RFL permits authoring Rust-based modules that are compiled and linked into the Linux kernel. Rust would be the only other general purpose language apart from C used to author Kernel modules. This is a very exciting milestone for Rust, but it's also a big challenge.
+
+Integrating Rust into the Linux kernel means that Rust must be able to interoperate with the kernel's low-level C primitives for things like locking, linked lists, allocation, and so forth.
+Integration requires Rust to expose low-level capabilities that don't all have stable interfaces.
+In some cases, the needed features may be stable, but may not be enough to provide for an ergonomic coding style (e.g., see the macros used for [pinned (in-place) initialization][pinned-init]).
+
+[pinned-init]: https://rust-for-linux.com/pinned-init
+[arclk]: https://rust-for-linux.com/arc-in-the-linux-kernel
+
+In the short term, the biggest blocker to the RFL exiting "experimental" status is its use of unstable features. Because unstable features have no kind of reliability guarantee, this in turn means that RFL can only be built with a specific, pinned version of the Rust compiler. This is a challenge for distributions which wish to be able to build a range of kernel sources with the same compiler, rather than having to select a particular toolchain for a particular kernel version.
+
+Longer term, having Rust in the Linux kernel is an opportunity to expose more C developers to the benefits of using Rust. But that exposure can go both ways. If Rust is constantly causing pain related to toolchain instability, or if Rust isn't able to interact gracefully with the kernel's data structures, kernel developers may have a bad first impression that causes them to write off Rust altogether. We wish to avoid that outcome. And besides, the Linux kernel is exactly the sort of low-level systems application we want Rust to be great for!
+
+For deeper background, please refer to these materials:
```

Comment:
Two of the links @fbq suggested are particularly relevant:

  - The article on the latest Maintainer Summit: [Committing to Rust for kernel code](https://lwn.net/Articles/952029/)
  - The LWN index on articles related to Rust in the kernel: https://lwn.net/Kernel/Index/#Development_tools-Rust

Also a couple talks:

  - [Our latest status update at LPC](https://www.youtube.com/watch?v=qvlgIaYrd3g).
  - [Linus on Rust](https://www.youtube.com/watch?v=OvuEYtkOH88&t=335s).